src/_pytest/main.py

```diff
@@ -367,6 +365,35 @@ def pytest_runtestloop(session: Session) -> bool:
     return True
 
 
+def _is_setuptools_project(path: Path) -> bool:
+    """Attempt to detect if ``path`` is the root of a setuptools project by
+    checking the existence of setup.py, setup.cfg, or pyproject.toml.
+    """
+    indicators = ("setup.py", "setup.cfg")
+    if any((path / f).exists() for f in indicators):
+        return True
+
+    toml_file = path / "pyproject.toml"
+    if toml_file.is_file():
+        if sys.version_info >= (3, 11):
+            import tomllib
+        else:
+            import tomli as tomllib
+
+        toml_text = toml_file.read_text(encoding="utf-8")
+        try:
+            config = tomllib.loads(toml_text)
+        except tomllib.TOMLDecodeError as exc:
+            raise UsageError(f"{toml_file}: {exc}") from exc
+
+        result = config.get("build-system", {}).get("requires", {})
```

Comment:
I recommend using a more verbose variable name, one that is not generic but corresponds to the data being assigned to it. This is because we should optimize for readability, since any code is read much more often than it's written.

Reply:
Noted, for now this code has been removed